apis/core.oam.dev/v1alpha2/core_scope_types.go

```diff
@@ -47,8 +47,24 @@ type HealthScopeSpec struct {
 	// ProbeInterval is the amount of time in seconds between probing tries.
 	ProbeInterval *int32 `json:"probe-interval,omitempty"`
 
+	// AppRefs records references of applications' components
+	AppRefs AppReferences `json:"appReferences,omitempty"`
+
 	// WorkloadReferences to the workloads that are in this scope.
-	WorkloadReferences []corev1.ObjectReference `json:"workloadRefs"`
+	// +deprecated
+	WorkloadReferences []corev1.ObjectReference `json:"workloadRefs,omitempty"`
+}
+
+// AppReferences records references of applications' components
+type AppReferences map[string]CompReferences
+
+// CompReferences records references of components' resources
+type CompReferences map[string]ComponentReference
+
```

Comment:
Why a new type `CompReferences` is added?

Reply:
CRD map type must use a named type as value type.